### PR TITLE
BO-47 Migrate to Hiera 5 and 3.3 codebase

### PR DESCRIPTION
**Pull request by kris-lab**
Part of https://github.com/cargomedia/puppet-cargomedia/issues/1795

### CODE REVIEW

**Comment by kris-lab** [comment]
@ppp0 as discussed please review

File: modules/puppetserver/manifests/environment.pp [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+    content => template("${module_name}/puppet/hiera.yaml"),
+    group   => '0',
+    owner   => '0',
+    mode    => '0644',
+    before  => Package['puppetserver'],
+    notify  => Service['puppetserver'],
+  }
+
+  file { "${data_directory}/common.yaml":
+    ensure  => file,
```
Comment (ppp0) [inline modules/puppetserver/manifests/environment.pp]: How about we use one `file` declaration and group all resources? e.g.
```pp
file { 'foo':
  ensure => file;
 'bar':
  ensure => directory;
[..]
}
```

**Comment by ppp0** [comment]
* [this file: (modules/puppet/lib/puppet/parser/functions/hiera_lookup.rb)](https://github.com/cargomedia/puppet-packages/blob/master/modules/puppet/lib/puppet/parser/functions/hiera_lookup.rb) not needed anymore? Where's its usage?

**Comment by kris-lab** [comment]
> modules/puppet/lib/puppet/parser/functions/hiera_lookup.rb) 

No idea why this was implemented. @tomaszdurka do you remember anything about that?

**Comment by kris-lab** [comment]
> http://localhost:8080/v2/nodes/' + certname + '/facts

We use hardcoded url with end-point which is invalid on production. `puppetdb` listens on port 8081.

**Comment by kris-lab** [comment]
@ppp0 please re-review

**Comment by ppp0** [comment]
lgtm! let's wait for the build to finish

**Pull request merged**